docs/src/main/asciidoc/security-iap.adoc

```diff
@@ -13,8 +13,11 @@ The following claims are validated automatically:
  * Expiration time
  * Issuer
 
-Additionally, Audience (`"aud"`) validation is enabled when the application is running on AppEngine/ComputeEngine/Kubernetes, or when a custom audience is provided through `spring.cloud.gcp.security.iap.audience` property.
-The custom property overrides automatic environment-based checks.
+Additionally, Audience (`"aud"`) validation is enabled when the application is running on AppEngine, or when a custom audience is provided through `spring.cloud.gcp.security.iap.audience` property.
+The custom property, if available, overrides the automatic AppEngine check.
+
+IMPORTANT: The audience check is *not* enabled by default on Compute Engine or Kubernetes Engine.
+To fully secure an application running on Compute Engine or Kubernetes Engine, find the Audience string per instructions the https://cloud.google.com/iap/docs/signed-headers-howto#verify_the_jwt_payload[Verify the JWT payload] guide, and specify it in `spring.cloud.gcp.security.iap.audience` property.
```

Comment:
If not checking the audience is a security vulnerability, we should not allow this by default. It's a pity that we can't determine it automatically though.

Reply:
I am on the fence about this. On the one hand, this is more of a backup, defense-in-depth mechanism -- IAP proxy itself will validate all claims before routing to the application. On the other hand, if someone does figure out how to bypass IAP, not validating audience means one can get a [valid, recent, GCP-signed] IAP token from any application and it will be accepted.

Reply:
For posterity -- we are removing GCE/GKE audience autoconfiguration for now, and reaching out to IAP team for guidance.